llmc/encoder.cuh

```diff
@@ -17,7 +17,7 @@ In the backward pass, the gradients flow to both, handled by different kernels
 // CUDA kernels
 
 __global__ void encoder_forward_kernel3(floatX* out,
-                               const int* inp, const floatX* wte, const floatX* wpe,
+                               const int* inp, const floatX* wte, const floatX* wpe, int use_rope,
```

Comment:
I don't like this because `use_rope` has nothing to do with this encoding function, which we want to be nice and modular and self-contained. Maybe something like `use_positional` or something like that.

Reply:
agree, can refactor a few of those bits quickly